src/test/java/org/tarantool/cluster/ClusterServiceStoredFunctionDiscovererIT.java

```diff
@@ -186,9 +200,68 @@ public void testFunctionWithError() {
         control.openConsole(INSTANCE_NAME).exec(functionCode);
 
         TarantoolClusterStoredFunctionDiscoverer discoverer =
-                new TarantoolClusterStoredFunctionDiscoverer(clusterConfig, client);
+            new TarantoolClusterStoredFunctionDiscoverer(clusterConfig, client);
 
         assertThrows(TarantoolException.class, discoverer::getInstances);
     }
 
+    @Test
+    @DisplayName("fetched a subset of valid addresses")
+    public void testFilterBadAddressesData() {
+        final List<String> allHosts = Arrays.asList(
+            "host1:3313",
+            "host:abc",
+            "192.168.33.90",
+            "myHost",
+            "10.30.10.4:7814",
+            "host:311:sub-host",
+            "instance-2:",
+            "host:0",
+            "host:321981"
+        );
+
+        final Set<String> expectedFiltered = new HashSet<>(
+            Arrays.asList(
+                "host1:3313",
+                "192.168.33.90",
+                "myHost",
+                "10.30.10.4:7814"
+            )
+        );
+
+        String functionCode = makeDiscoveryFunction(ENTRY_FUNCTION_NAME, allHosts);
+        control.openConsole(INSTANCE_NAME).exec(functionCode);
+
+        TarantoolClusterStoredFunctionDiscoverer discoverer =
+            new TarantoolClusterStoredFunctionDiscoverer(clusterConfig, client);
+
+        Set<String> instances = discoverer.getInstances();
+
+        assertNotNull(instances);
+        assertFalse(instances.isEmpty());
+        assertEquals(expectedFiltered, instances);
+    }
+
+    @Test
+    @DisplayName("fetched an empty set after filtration")
+    public void testFullBrokenAddressesList() {
```

Comment:
This case is interesting. What we'll do if the resulting address list is empty? I think we should report this to logs (when we'll implement them) and discard the result (I mean don't disconnect from a current instance).

Reply:
Add a warning is a good idea. The empty list will be skipped by the cluster client [here](https://github.com/tarantool/tarantool-java/blob/master/src/main/java/org/tarantool/TarantoolClusterClient.java#L276)

Reply:
I guess it worth to implement reporting in the scope of #194, so I marked this as resolved. Thanks for the clarification.